packages/phone-number-privacy/signer/src/domain/domain.service.ts

```diff
@@ -1,83 +1,221 @@
+      const domainState = await getDomainState(domain, logger)
+      let resultResponse: DomainStatusResponse
+      if (domainState) {
+        resultResponse = {
+          domain,
+          counter: domainState[DOMAINS_STATES_COLUMNS.counter]!,
+          disabled: domainState[DOMAINS_STATES_COLUMNS.disabled]!,
+          timer: domainState[DOMAINS_STATES_COLUMNS.timer]!,
         }
-      })
-      .catch((e) => {
-        logger.error('Error while disabling domain', e)
-        respondWithError(
-          Endpoints.DISABLE_DOMAIN,
-          response,
-          500,
-          ErrorMessage.DATABASE_UPDATE_FAILURE
-        )
-      })
+      } else {
+        resultResponse = {
+          domain,
+          counter: 0,
+          disabled: false,
+          timer: 0,
+        }
+      }
+      response.status(200).send(resultResponse)
+    } catch (error) {
+      logger.error('Error while getting domain status', error)
+      respondWithError(
+        Endpoints.DOMAIN_QUOTA_STATUS,
+        response,
+        500,
+        ErrorMessage.DATABASE_GET_FAILURE
+      )
+    }
   }
 
-  public async handleGetDomainStatus(
+  public async handleGetBlindedMessagePartialSig(
     request: Request<{}, {}, DomainRestrictedSignatureRequest>,
     response: Response
   ): Promise<void> {
     const logger = response.locals.logger
-    logger.info('Getting domain status', { domain: request.body.domain })
-    getDomainState(request.body.domain, logger)
-      .then((result) => {
-        let resultResponse: DomainStatusResponse
-        if (result) {
-          resultResponse = {
-            domain: result[DOMAINS_STATES_COLUMNS.domain],
-            counter: result[DOMAINS_STATES_COLUMNS.counter],
-            disabled: result[DOMAINS_STATES_COLUMNS.disabled],
-            timer: result[DOMAINS_STATES_COLUMNS.timer],
-          }
-        } else {
-          resultResponse = {
-            domain: request.body.domain,
-            counter: 0,
-            disabled: false,
-            timer: 0,
-          }
-        }
-        response.status(200).send(resultResponse)
+    const domain = request.body.domain
+    if (!this.authAndCheckKnownDomain(domain, response, Endpoints.DOMAIN_SIGN, logger)) {
+      logger.warning('Cannot handle request')
+      return
+    }
+    logger.info('Processing request to get domain signature ', {
+      name: domain.name,
+      version: domain.version,
+      hash: domainHash(domain),
+    })
+
+    const trx = await getTransaction()
+    const domainState = (await getDomainStateWithLock(domain, trx, logger))! // TODO: what if we dont have domain state yet
+    // @Alec we never actually create any domain state in first place.
+    const doesHaveRemainingQuota = await this.quotaService.doesHaveRemainingQuota(
+      domain,
+      domainState
+    )
+    if (!doesHaveRemainingQuota) {
+      trx.rollback()
+      logger.warn(`Exceeded quota`, {
+        name: domain.name,
+        version: domain.version,
```

Comment:
Adding `hash` here is helpful to track down any errors.
```suggestion
        version: domain.version,
        hash: domainHash(domain),
```